vmm/src/vm_config.rs

```diff
@@ -1056,4 +1065,45 @@ impl VmConfig {
             self.cpus.max_vcpus
         }
     }
+
+    pub fn consume_fds(&mut self, fds: Vec<i32>) -> Result<(), PayloadConfigError> {
+        // Update self.external_fds, if appropriate.
+        let Some(external_fds) = self.external_fds.as_mut() else {
+            if fds.is_empty() {
+                return Ok(());
+            } else {
+                return Err(PayloadConfigError::FdCountMismatch);
+            }
+        };
+
+        if external_fds.ids.len() != fds.len() {
```

Comment:
why does the number of provided component (device) IDs must match the number of FDs?! This sounds wrong.

Reply:
That's the design of the proposed API syntax. I agree it should be explicitly explained, say from the commit message.